bambam.py

```diff
@@ -216,10 +224,18 @@ def print_letter(self, char):
         textpos.centery = h
         self.screen.blit(text, textpos)
 
-    def glob_data(self, pattern):
+    def glob_dir(self, path, pattern):
+        files = glob.glob(os.path.join(path, pattern))
+        dirs = [x for x in glob.glob(os.path.join(path, '*')) if os.path.isdir(x)]
+        for subdir in dirs:
+            files.extend(self.glob_dir(subdir, pattern))
+        return files
+
+    def glob_data(self, patterns):
         fileList = []
         for dataDir in self.dataDirs:
-            fileList.extend(glob.glob(os.path.join(dataDir, pattern)))
+            for pattern in patterns:
+                fileList.extend(self.glob_dir(dataDir, pattern))
```

Comment:
Could you also make `glob_dir` take multiple patterns? Otherwise we have to walk each tree separately for every supported file extension.

Reply:
Glob doesn't work cleanly with multiple patterns, so passing multiple would change the order of tree-walks but not its number. I just submitted a new version without glob that only does one pass through the directory trees. It also supports case-insensitive file name matching. Would you say that's a preferable change?